### PR TITLE
fix(cli): use path name for classname in JUnit reports instead of request URL

### PR DESCRIPTION
**Pull request by sharan-bruno**
**Jira link: https://usebruno.atlassian.net/browse/BRU-3123**

### Description

The CLI's JUnit reporter used the request URL as the classname on each <testcase>. Since URLs change per environment/region, the same test produced duplicate entries in CI tools (Xray, Azure DevOps) that key tests by classname + name.

This changes classname to use the request name (matching Postman), so the same test maps to a single test case across all environments.

packages/bruno-cli/src/reporters/junit.js — classname now uses result.name instead of result.request.url
Added test coverage asserting classname is the request name
#### Contribution Checklist:

After fix:
image

- [ ] **I've used AI significantly to create this pull request**
- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **Bug Fixes**
  * JUnit test reports now derive each testcase's classname from the test result path (file/path-derived name) instead of the request URL, improving report clarity.

* **Tests**
  * Added a unit test to assert classname derivation and updated JUnit test snapshots across platforms to reflect the new classname values.

### CODE REVIEW

File: packages/bruno-cli/src/reporters/junit.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ const makeJUnitOutput = async (results, outputPath) => {
     const testCount = result.testResults ? result.testResults.length : 0;
     const postResponseTestCount = result.postResponseTestResults ? result.postResponseTestResults.length : 0;
     const totalTests = assertionTestCount + preRequestTestCount + testCount + postResponseTestCount;
+    const classname = result.name;
```
Comment (sanish-bruno) [inline packages/bruno-cli/src/reporters/junit.js]: have we decided on what should be the classname? path or the name?

**Review by sanish-bruno — CHANGES_REQUESTED** [review]

File: packages/bruno-cli/src/reporters/junit.js [hunk 2]
```diff
@@ -34,7 +35,7 @@ const makeJUnitOutput = async (results, outputPath) => {
       const testcase = {
         '@name': `${assertion.lhsExpr} ${assertion.rhsExpr}`,
         '@status': assertion.status,
-        '@classname': result.request.url,
+        '@classname': classname,
```
Comment (sanish-bruno) [inline packages/bruno-cli/src/reporters/junit.js:39]: we need to update the snapshots fro junit playwright tests we have

**Review by sanish-bruno — CHANGES_REQUESTED** [review]

File: packages/bruno-cli/src/reporters/junit.js (hunk 1 above)
Comment (sharan-bruno) [reply packages/bruno-cli/src/reporters/junit.js]: updated it to path

File: packages/bruno-cli/src/reporters/junit.js (hunk 2 above)
Comment (sharan-bruno) [reply packages/bruno-cli/src/reporters/junit.js:39]: updated it

File: tests/runner/collection-run-report/collection-run-report.spec.ts [hunk 3]
```diff
@@ -13,8 +13,9 @@ function normalizeJunitReport(xmlContent: string): string {
     .replace(/time="[^"]*"/g, 'time="0.100"')
     // Replace file paths with normalized path
     .replace(/file="[^"]*[\\/][^"]*"/g, 'file="/mock/path/to/file.bru"')
-    // Replace test paths with normalized path
-    .replace(/classname="[^"]*[\\/][^"]*"/g, 'classname="/test/path/collection"');
+    // classname is the request's relative path within the collection (stable across
+    // environments); only normalize the path separator so snapshots match on all platforms
+    .replace(/classname="([^"]*)"/g, (_match, value) => `classname="${value.replace(/\\/g, '/')}"`);
```
Comment (sanish-bruno) [inline tests/runner/collection-run-report/collection-run-report.spec.ts]: check if this regex works fine! it seems like it is different from the prev version of it `/classname="[^"]*[\\/][^"]*"/g`

**Review by sanish-bruno — CHANGES_REQUESTED** [review]

File: tests/runner/collection-run-report/collection-run-report.spec.ts (hunk 3 above)
Comment (sharan-bruno) [reply tests/runner/collection-run-report/collection-run-report.spec.ts]: Verified — works correctly, the change is intentional. classname is stripExtension(result.path), the stable relative request path. The old regex only matched paths containing a separator and swapped in a mock; the new one keeps the real path and just normalizes \→/ so Windows matches the snapshot. Confirmed all three platform snapshots store classname="api/v1/users" with forward slashes, so it lines up. Quote-bounded [^"]* + g flag handle the multiple/edge cases fine.

Comment (bijin-bruno) [reply tests/runner/collection-run-report/collection-run-report.spec.ts]: In this case, i think we should try to keep the realistic value for file as well for consistency. 
Can we avoid this completely? Windows report will be validated only in a win environment I believe. Should not create any issues with path normalization.

**Review by bijin-bruno — CHANGES_REQUESTED** [review]

File: tests/runner/collection-run-report/collection-run-report.spec.ts (hunk 3 above)
Comment (sharan-bruno) [reply tests/runner/collection-run-report/collection-run-report.spec.ts]: update classname in JUnit report to use backslashes for Windows compatibility

**Review by bijin-bruno — APPROVED** [review]

**Review by sanish-bruno — APPROVED** [review]

**Review by bijin-bruno — APPROVED** [review]
